### PR TITLE
Adds new Role restricted Traitor item: Reticence Assasination Exosuit

### PR DESCRIPTION
**Pull request by ChuChu42**
## About The Pull Request

Adds a decade old item that has been in the code for a LONG LONG TIME but yet still unobtainable without admin intervention

as a traitor item as it was (supposedly?) designed as

screenshots of the item
image
image

and a ameteur demonstration video

https://github.com/tgstation/tgstation/assets/118483925/641dfc9c-d548-4e56-a106-6f85a5c12ad4

Note: this is my first ever addition PR in tgstation, please inform me if you believe i have made any issues!
## Why It's Good For The Game

makes a decade old mime-themed item, obtainable by mime traitors
this item is a **VERY FRAGILE** silent, fast, "assasination" exosuit
the suit has barely any armor and exactly 100 durability making it extremely fragile to oncoming attacks, however, it includes a RCD, a silenced pistol and the ability to punch open structures

effectively making this a glass cannon

personally i think this item should have been obtainable by mime traitors in the first place, but i dont get why it never was, i believe it to be a fairly balanced item for any ordinary traitor to use ((unlike certain items))

so i believe its a good addition
## Changelog
:cl:
add: Added Reticence Assasination Exosuit For 20 TC to mime traitors
tweak: makes the mime mech have a air tank by default
tweak: reticence is faster and slightly stronger
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
It looks like you've used a lot of spaces for indentation instead of tabs, you'll need to replace those with tabs.

**Comment by K4rlox** [comment]
> It looks like you've used a lot of spaces for indentation instead of tabs, you'll need to replace those with tabs.

i already fixed those and it worked for me when i built it, can you point out which?

**Comment by K4rlox** [comment]
> It looks like you've used a lot of spaces for indentation instead of tabs, you'll need to replace those with tabs.

ITS ALL TABS YET ITS STILL FAILING WHAT THE FUCK DO I DO

**Comment by K4rlox** [comment]
should work

**Comment by moocowswag** [comment]
How does reloading the gun work, does it just use energy?

Another thing is using this in the long term would require the mime to have a good understanding of robotics which I'm personally okay with but this is a big leap in the mime's need to know what they are doing, gathering parts to upgrade the mech, making a safe place to recharge it, etc, Ive done the same as a clown esp when I oddy / honk mech murderboned but I dont expect the avg player to be capable of that.

So to me, this seems more like a counterpart to the clown car in that most players wont maintain the mech and will mainly just use it to get as many kills as they can quickly...except it can be hacked and anyone can steal it thanks to the mech rework...

File: code/modules/uplink/uplink_items/job.dm [hunk 1]
```diff
@@ -369,3 +369,14 @@
 	restricted_roles = list(JOB_ASSISTANT, JOB_MIME, JOB_CLOWN)
 	restricted = TRUE
 	refundable = FALSE
+
+
+/datum/uplink_item/role_restricted/reticence
+	name = "Reticence Cloaked Assasination exosuit"
+	desc = "A silent, fast, and nigh-invisible but exepctionally fragile miming exosuit! \
```
Comment (Vincent983) [inline code/modules/uplink/uplink_items/job.dm:376]: It's spelled as "exceptional", also I don't believe you need to capitalize "Near-Silenced"

**Comment by deathrobotpunch** [comment]
a mech seems pretty OP but I would be down for a completely invisible modsuit

**Comment by deathrobotpunch** [comment]
> a mech seems pretty OP but I would be down for a completely invisible modsuit (only the modsuit being invisible)

**Comment by necromanceranne** [comment]
The reticence, despite being near invisible, is also made of tinfoil and will die extremely quickly to a few laser shots. Its main draw is complete camouflage, but once you spot it, the pilot will either have to make sure you die immediately, or you can mostly put it down with the majority of station lethal weaponry.

20 TC makes sense for what it is capable of doing, and for the drawbacks. (though I think it needs a reputation lock)

**Comment by githubuser4141** [comment]
20 TC seems like a lot for something that is this frail. It'd suck to buy this to experiment with or for a gimmick to have it wrecked a few minutes later from either bad luck (mobs target it) or a few stray laser shots. Maybe it could be non-targetable by non-megafauna mobs, or it refunds some amount of your TC after it's destroyed?

**Comment by EricZilla** [comment]
If its so weak people dont buy it, then we can buff it later, but being wholesale invisible and being able to rip open the floor is kinda nuts

**Comment by Vincent983** [comment]
> 20 TC seems like a lot for something that is this frail. It'd suck to buy this to experiment with or for a gimmick to have it wrecked a few minutes later from either bad luck (mobs target it) or a few stray laser shots. Maybe it could be non-targetable by non-megafauna mobs, or it refunds some amount of your TC after it's destroyed?
> 
> What stock parts does it start with - iirc Tier 1 & 2 have very poor endurance which will be confusing for someone who doesn't have in-depth knowledge of the mech system, or when Sci/Robotics/Mining isn't available.

it seems to start with tier 3 parts

**Comment by necromanceranne** [comment]
> it shouldnt come with the mounted rcd just go to robotics to print to add equipment mech can equip bout everything anywahs

it's meant to replicate the mime's ability to build walls.

Personally, I think it should just print invisible walls, not actual walls.

**Comment by SomeoneYouProbablyKnow** [comment]
> a mech seems pretty OP but I would be down for a completely invisible modsuit

I think a mech makes sense for this specifically (and i'm a mech hater). Since it has its own health and a very specific way to heal. Can't just pump on chems and be invisible on top of it.

> Another thing is using this in the long term would require the mime to have a good understanding of robotics <...>

Should probably reflect these difficulties in the flavour text. 

> except it can be hacked and anyone can steal it if youre caught out of it thanks to the mech rework.

Can always lock it to mime's vow, right? just prevent people without it piloting or even punishing them in some funny way (permanent monochromia)

File: code/modules/uplink/uplink_items/job.dm (hunk 1 above)
Comment (K4rlox) [reply code/modules/uplink/uplink_items/job.dm:376]: fixed

**Comment by K4rlox** [comment]
> > 20 TC seems like a lot for something that is this frail. It'd suck to buy this to experiment with or for a gimmick to have it wrecked a few minutes later from either bad luck (mobs target it) or a few stray laser shots. Maybe it could be non-targetable by non-megafauna mobs, or it refunds some amount of your TC after it's destroyed?
> > What stock parts does it start with - iirc Tier 1 & 2 have very poor endurance which will be confusing for someone who doesn't have in-depth knowledge of the mech system, or when Sci/Robotics/Mining isn't available.
> 
> it seems to start with tier 3 parts

tier 3 parts and tier 4 cell

**Comment by K4rlox** [comment]
> How does reloading the gun work, does it just use energy?
> 
> Another thing is using this in the long term would require the mime to have a good understanding of robotics which I'm personally okay with but this is a big leap in the mime's need to know what they are doing, gathering parts to upgrade the mech, making a safe place to recharge it, etc since they are antagonistic they wont be able to use the station's facilities very easily and would need alot of prep.
> 
> So to me, this seems more like a counterpart to the clown car in that most players wont maintain the mech and will mainly just use it to get as many kills as they can quickly...except it can be hacked and anyone can steal it if youre caught out of it thanks to the mech rework.
> 
> There is very different playstyle with this, its a good addition to the mime uplink since it changes the way you might approach the game, its just not friendly to alot of players and I'm afraid that might be used as an excuse to hold it back.

the mime gun has 999 ammo and is reloadable, however, unlike clown car where "get as many kills as possible and get out" you are not able to murderbone as mime mech since its not good against group enemies


i will slightly update the reticence's armor and health to make it be a glass cannon instead of a tinfoil cannon tough

**Comment by K4rlox** [comment]
> The reticence, despite being near invisible, is also made of tinfoil and will die extremely quickly to a few laser shots. Its main draw is complete camouflage, but once you spot it, the pilot will either have to make sure you die immediately, or you can mostly put it down with the majority of station lethal weaponry.
> 
> 20 TC makes sense for what it is capable of doing, and for the drawbacks. (though I think it needs a reputation lock)

i tweaked the mech to be made out of glass rather than tinfoil now, it also has incrased speed and made it so destroying the mech stuns their pilot for 1 second instead of 40 so they can just immidently bail as soon as the mech gets destroyed


i whould like to also mention

the mech wreckage is ALSO transparent
image

**Comment by K4rlox** [comment]
if maintainers think that i buffed the mech too much feel free to point it out

**Comment by jughu** [comment]
> > it shouldnt come with the mounted rcd just go to robotics to print to add equipment mech can equip bout everything anywahs
> 
> it's meant to replicate the mime's ability to build walls.
> 
> Personally, I think it should just print invisible walls, not actual walls.

im not saying it because giving a way to remove rwalls from any range is a very strong ability especially as a invinsible mech it could be very abusable but the idea of having a rcd that prints temporsry invinsible walls sounds funnier

**Comment by K4rlox** [comment]
> > > it shouldnt come with the mounted rcd just go to robotics to print to add equipment mech can equip bout everything anywahs
> > 
> > 
> > it's meant to replicate the mime's ability to build walls.
> > Personally, I think it should just print invisible walls, not actual walls.
> 
> im not saying it because giving a way to remove rwalls from any range is a very strong ability especially as a invinsible mech it could be very abusable but the idea of having a rcd that prints temporsry invinsible walls sounds funnier

im not skilled enough to code that *cries*

**Comment by MoonDeer728** [comment]
What you can say about 4 toolbox hits

**Comment by K4rlox** [comment]
> What you can say about 4 toolbox hits

i buffed melee armor a little

also it can just run away

**Comment by ATH1909** [comment]
ah, this takes me back (https://github.com/tgstation/tgstation/pull/44612)

> the mime gun has 999 ammo and is reloadable, however, unlike clown car where "get as many kills as possible and get out" you are not able to murderbone as mime mech since its not good against group enemies

is it possible to make the gun just have infinite (or energy-based) ammo, then?

**Comment by ATH1909** [comment]
> im not saying it because giving a way to remove rwalls from any range is a very strong ability especially as a invinsible mech it could be very abusable but the idea of having a rcd that prints temporsry invinsible walls sounds funnier

the RCD animation gives away your presence ("hey, why is this wall deconstructing? I don't see anyone around..."), and as someone who's used it, the reticence heavily relies on people not knowing it exists and thus not squinting hard enough to see it.

**Comment by K4rlox** [comment]
> > im not saying it because giving a way to remove rwalls from any range is a very strong ability especially as a invinsible mech it could be very abusable but the idea of having a rcd that prints temporsry invinsible walls sounds funnier
> 
> the RCD animation gives away your presence ("hey, why is this wall deconstructing? I don't see anyone around..."), and as someone who's used it, the reticence heavily relies on people not knowing it exists and thus not squinting hard enough to see it. I think it's fine.

im pretty sure people will just think its a ninja with a rcd or something lol

**Pull request merged**